File: CS/SyncWithOutlook/SyncronizationOptionForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace SyncWithOutlook {
    public partial class SyncronizationOptionForm : Form {
        public static bool AllowCreateAppointmentInScheduler;
        public static bool AllowUpdateAppointmentInScheduler;
        public static bool AllowRemoveAppointmentInScheduler;

        public static bool AllowCreateAppointmentInOutlook;
        public static bool AllowUpdateAppointmentInOutlook;
        public static bool AllowRemoveAppointmentInOutlook;

        public SyncronizationOptionForm() {
            InitializeComponent();
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            this.Close();
        }

        private void SyncronizationOptionForm_Load(object sender, EventArgs e) {
            checkEditAllowCreateOutllok.EditValue = SyncronizationOptionForm.AllowCreateAppointmentInOutlook;
            checkEditAllowCreateScheduler.EditValue = SyncronizationOptionForm.AllowCreateAppointmentInScheduler;
            checkEditAllowRemoveOutlook.EditValue = SyncronizationOptionForm.AllowRemoveAppointmentInOutlook;
            checkEditAllowRemoveScheduler.EditValue = SyncronizationOptionForm.AllowRemoveAppointmentInScheduler;
            checkEditAllowUpdateOutlook.EditValue = SyncronizationOptionForm.AllowUpdateAppointmentInOutlook;
            checkEditAllowUpdateScheduler.EditValue = SyncronizationOptionForm.AllowUpdateAppointmentInScheduler;
        }

        protected override void OnClosing(CancelEventArgs e) {
            SyncronizationOptionForm.AllowCreateAppointmentInOutlook = (bool)checkEditAllowCreateOutllok.EditValue;
            SyncronizationOptionForm.AllowCreateAppointmentInScheduler = (bool)checkEditAllowCreateScheduler.EditValue;
            SyncronizationOptionForm.AllowRemoveAppointmentInOutlook = (bool)checkEditAllowRemoveOutlook.EditValue;
            SyncronizationOptionForm.AllowRemoveAppointmentInScheduler = (bool)checkEditAllowRemoveScheduler.EditValue;
            SyncronizationOptionForm.AllowUpdateAppointmentInOutlook = (bool)checkEditAllowUpdateOutlook.EditValue;
            SyncronizationOptionForm.AllowUpdateAppointmentInScheduler = (bool)checkEditAllowUpdateScheduler.EditValue;
            base.OnClosing(e);
        }
    }
}

File: VB/SyncWithOutlook/SyncronizationOptionForm.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace SyncWithOutlook

    Public Partial Class SyncronizationOptionForm
        Inherits Form

        Public Shared AllowCreateAppointmentInScheduler As Boolean

        Public Shared AllowUpdateAppointmentInScheduler As Boolean

        Public Shared AllowRemoveAppointmentInScheduler As Boolean

        Public Shared AllowCreateAppointmentInOutlook As Boolean

        Public Shared AllowUpdateAppointmentInOutlook As Boolean

        Public Shared AllowRemoveAppointmentInOutlook As Boolean

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Close()
        End Sub

        Private Sub SyncronizationOptionForm_Load(ByVal sender As Object, ByVal e As EventArgs)
            checkEditAllowCreateOutllok.EditValue = AllowCreateAppointmentInOutlook
            checkEditAllowCreateScheduler.EditValue = AllowCreateAppointmentInScheduler
            checkEditAllowRemoveOutlook.EditValue = AllowRemoveAppointmentInOutlook
            checkEditAllowRemoveScheduler.EditValue = AllowRemoveAppointmentInScheduler
            checkEditAllowUpdateOutlook.EditValue = AllowUpdateAppointmentInOutlook
            checkEditAllowUpdateScheduler.EditValue = AllowUpdateAppointmentInScheduler
        End Sub

        Protected Overrides Sub OnClosing(ByVal e As CancelEventArgs)
            AllowCreateAppointmentInOutlook = CBool(checkEditAllowCreateOutllok.EditValue)
            AllowCreateAppointmentInScheduler = CBool(checkEditAllowCreateScheduler.EditValue)
            AllowRemoveAppointmentInOutlook = CBool(checkEditAllowRemoveOutlook.EditValue)
            AllowRemoveAppointmentInScheduler = CBool(checkEditAllowRemoveScheduler.EditValue)
            AllowUpdateAppointmentInOutlook = CBool(checkEditAllowUpdateOutlook.EditValue)
            AllowUpdateAppointmentInScheduler = CBool(checkEditAllowUpdateScheduler.EditValue)
            MyBase.OnClosing(e)
        End Sub
    End Class
End Namespace
